### ISSUE TITLE
Light switcher

### ISSUE THREAD

**Issue by metehan** [CoenraadS/Bracket-Pair-Colorizer-2#62]
Hi, It would be great if you could provide option to quickly switch color schema. In mornings I am using light theme and yellow brackets becomes invisible.

**Comment by ellipticaldoor** [comment]
Same

**Comment by giorgio79** [comment]
Yeah, the module works great out of the box for the dark theme. I woud love a selector that changes preconfigged colors for the light theme as well. Perhaps it could detect "light" themes as well and switch.

**Comment by yaagma** [comment]
It would be nice if colors could be set by theme. 
For example, with "editor.tokenColorCustomizations", you can do this:

-----------
   "editor.tokenColorCustomizations": {
        "[Default Light+]": {
            "comments": "#6d7688"
        },
        "[Nord]": {
            "comments": "#xxxxxx"
        }
    },
----------------

In the same way, it would be very useful something like this:

"bracket-pair-colorizer-2.colors": {
       "[Default Light+]": {
            ...
        },
        "[Nord]": {
            ....
        }
}

Thanks
PS: Excuse my english

**Comment by OskarKlintrot** [comment]
Support for Light theme out of the box would be awesome!

**Comment by shmulvad** [comment]
Would also really love this. If not what @yaagma described, then something that integrates with a plugin like [Day Night Theme Switcher](https://github.com/freetonik/vscode-dnts).

**Comment by sldsrg** [comment]
As a temporary workaround, we can use terminal colors from theme:
```
  "bracket-pair-colorizer-2.colors": [
    "terminal.ansiBrightMagenta",
    "terminal.ansiYellow",
    "terminal.ansiBrightCyan"
  ]
```

**Comment by sldsrg** [comment]
Another workaround:

```
// settings.json
{
  "workbench.colorCustomizations": {
    "[Default Light+]": {
      // define custom colors for Light+ theme
      "colorizer.color-1": "#32CD32",
      "colorizer.color-2": "#FF00FF",
      "colorizer.color-3": "#1E90FF"
    },
    // define custom colors for other themes
    // (fallback! must be last in sequence)
    "colorizer.color-1": "#FFD700",
    "colorizer.color-2": "#DA70D6",
    "colorizer.color-3": "#87CEFA"
  },
  // use custom colors
  "bracket-pair-colorizer-2.colors": [
    "colorizer.color-1",
    "colorizer.color-2",
    "colorizer.color-3"
  ]
}
```

**Comment by macintacos** [comment]
I'd like to just note that @sldsrg's solution works perfectly (and works just fine with the Day Night Theme Switcher that @shmulvad mentioned earlier).

**Comment by shmulvad** [comment]
@sldsrg Your first workaround, using terminal colours, works for me.

I am however unable to use your second workaround. When I try to use that in my settings I get the following error:
```
Property colorizer.color-1 is not allowed.
Property colorizer.color-2 is not allowed.
Property colorizer.color-3 is not allowed.
```

I only get these errors for the definitions inside `[Default Light+]`, not for the ones in the top scope of `workbench.colorCustomizations`. Any ideas as to how to fix this?

**Comment by macintacos** [comment]
@shmulvad for what it's worth, I see the same errors but it still works fine. Possibly just a parsing issue on VSCode's part when trying to identify problems?

**Comment by VideoPlayerCode** [comment]
@sldsrg WTF, You are a GENIUS! I was reading your terminal solution and thought "Hmm, smart!". But then I saw your solution for literally creating custom colors per-theme and my mind is blown! THANK YOU!

**Comment by VideoPlayerCode** [comment]
Didn't like the suggested colors for light theme above, though...

So, after working on this for a while via https://www.hexcolortool.com/, I've now converted the default Bracket Pair Colorizer colors to very accurate analogues that look good on a light (white in my case) background. Basically, with these settings you get the same familiar bracket colors on both light and dark themes, but they're now more readable on light backgrounds.

```json
    "bracket-pair-colorizer-2.colors": [
        "colorizer.color-1",
        "colorizer.color-2",
        "colorizer.color-3"
    ],
    "workbench.colorCustomizations": {
        "[Dracula.min White]": {
            // Bracket Pair Colorizer overrides.
            "colorizer.color-1": "#ff9900", // Orange-brown...
            "colorizer.color-2": "#f500f5", // Pink...
            "colorizer.color-3": "#54b9f8", // Medium-Blue...
        },
        // Fallback to default colors for all themes that lack overrides.
        "colorizer.color-1": "#ffd700", // "Gold". (Yellow with slight Orange)
        "colorizer.color-2": "#da70d6", // "Orchid". (Pink) 
        "colorizer.color-3": "#87cefa", // "LightSkyBlue". (Light Blue)
    }
}
```

Anyone with light themes should try these. They're looking excellent on white backgrounds.

**Comment by Jordan-Eckowitz** [comment]
does anybody know how to do the same color customization for the `scopeLineCSS` property?

**Comment by thegoodsheppard** [comment]
I'm trying to implement sldsrg's solution but `"workbench.colorCustomizations"` complains that "Property colorizer.color-1 is not allowed" when it's scoped to a theme. It does work in the main scope though. Has anyone else seen this? 

For example, the first reference to Navy doesn't trigger the error but the second does:

```json
"workbench.colorCustomizations": {
  "colorizer.color-1": "Navy",
  "[Default Light+]": {
    "colorizer.color-1": "Navy",
  }
}

**Comment by martendebruijn** [comment]
@thegoodsheppard You have to put the fallback values at the end, then it will work. However, it will not get rid of the "Property colorizer.color-n is not allowed" error (of which I currently have 21 warnings)

```json
"workbench.colorCustomizations": {
  "[Default Light+]": {
    "colorizer.color-1": "Navy",
  },
  "colorizer.color-1": "Navy",
}
```
